v-next/hardhat/src/internal/builtin-plugins/network-manager/hook-handlers/network.ts

```diff
@@ -11,14 +12,18 @@ import type {
   NetworkConnection,
 } from "@ignored/hardhat-vnext/types/network";
 
-import { JsonRpcRequestModifier } from "../json-rpc-request-modifiers/json-rpc-request-modifier.js";
+import { deepClone } from "@ignored/hardhat-vnext-utils/lang";
+
+import { isJsonRpcResponse } from "../json-rpc.js";
+import { createHandlersArray } from "../request-handlers/hanlders-array.js";
 
 export default async (): Promise<Partial<NetworkHooks>> => {
-  // This map is necessary because Hardhat V3 supports multiple network connections, requiring us to track them
-  // to apply the appropriate modifiers to each request.
-  // When a connection is closed, it is removed from the map. Refer to "closeConnection" at the end of the file.
-  const jsonRpcRequestModifiers: Map<number, JsonRpcRequestModifier> =
-    new Map();
+  // This map is essential for managing multiple network connections in Hardhat V3.
+  // Since Hardhat V3 supports multiple connections, we use this map to track each one
+  // and associate it with the corresponding handlers array.
+  // When a connection is closed, its associated handlers array is removed from the map.
+  // See the "closeConnection" function at the end of the file for more details.
+  const requestHandlersPerConnection: Map<number, RequestHandler[]> = new Map();
```

Comment:
I think we should change it to this, so that the array can still be GC'd if the user doesn't call `close()`.

```suggestion
  const requestHandlersPerConnection: WeakMap<NetworkConnection, RequestHandler[]> = new Map();
```

Reply:
To be clear, we should still keep close, as it's still valuable to offer control to the users, but if the GC decides to collect the network connection, it should also collect its `RequestHandler[]`.